apps/api/src/app/integrations/usecases/create-integration/create-integration.usecase.ts

```diff
@@ -180,6 +180,14 @@ export class CreateIntegration {
 
       const isActiveAndChannelSupportsPrimary = command.active && CHANNELS_WITH_PRIMARY.includes(command.channel);
       if (isMultiProviderConfigurationEnabled && isActiveAndChannelSupportsPrimary) {
+        await this.disableNovuIntegration.execute({
+          channel: command.channel,
+          providerId: command.providerId,
+          environmentId: command.environmentId,
+          organizationId: command.organizationId,
+          userId: command.userId,
+        });
```

Comment:
Before calculating the primary and priority fields for the created integration we have to disable the Novu integration for the channel first.